packages/modules/vehicles/leaf/pycarwings2.py

```diff
@@ -0,0 +1,462 @@
+# Copyright 2016 Jason Horne
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
```

Comment:
Es ist leider nicht möglich einen Teil der Software unter eine Lizenz zu stellen. pycarwings2 kann aber auch über pip installiert werden. Dazu in requirements.txt `pycarwings2==2.14` ergänzen. Dann wird das Paket beim Starten installiert.

Reply:
pycarwings2 ist in V1.9 wie in meinem soc-draft eingebunden. Siehe hier:
https://github.com/snaptec/openWB/blob/master/modules/soc_leaf/pycarwings2.py
Wenn es in V1.9 genutzt wird, dann muss dafür ja früher mal ein OK eingeholt worden sein. Oder?

Reply:
Wenn ich auf die Zusammenfassung https://de.wikipedia.org/wiki/Apache-Lizenz#Eigenschaften schaue, dann sehe ich auch kein Problem, pycarwings2 weiterhin zu nutzen.

Reply:
Ja, in 1.9 ist das so gewesen. In software2 verfolgen wir eine anderen Linie. Hier wird die komplette Software unter eine Lizenz gestellt. 
Bitte probiere, pycarwings über pip zu installieren. Vielleicht hat sich damit auch gleich das Crypto.Cipher-Problem gelöst.
Die imports musst du dann wieder auf import pycarwings2 ändern.
In der requirements.txt pycarwings2==2.14 hinzufügen und dann mit sudo systemctl restart openwb2.service den Sercvice neu starten. Dann wird das Package installiert. Das siehst du dann auch im main.log.

Reply:
Die Apache Lizenz ist nach meinem Verständnis ein Copy-Left und kein Copyright. Der beschriebene Lizenz-Umweg mag für Testzwecke gehen und für einige Nutzer von V2 ein möglicher Workaround sein. Das ist aber keine allgemeine Lösung für OpenWB V2 Nutzer mit einem älteren Leaf oder NV200. Mit dem vorgeschlagenen Workaround sind viele überfordert. Wie müsste eine Lösung aussehen, die man mit einem normalen Softwareupdate aufspielen kann?

Das Crypto.Cipher-Problem überschaue ich noch nicht. In der V1.9 ist Crypto.Cypher anscheinend noch enthalten. Sonst würde pycarwings2 dort ja nicht laufen. 
Ich musste auf meinem PC mit Python 3.12 zusätzlich Pycryptodome mit "pip install pycryptodome" installieren, um Crypto.Cipher verfügbar zu haben. Inwieweit Crypto.Cipher bereits/noch auf der OpenWB mit V2 verfügbar ist, kann ich nicht sagen. Bitte prüfen.

Reply:
Was für ein Workaround? Die requirements.txt ist der Standardweg, um nach einem Softwareupdate fehlende Python-Packages zu installieren. Der Endbenutzer muss dafür gar nichts tun.

pycryptodome ist nicht vorhanden. Aber ist das überhaupt erforderlich, wenn man pycarwings über den Paketmanager installiert?

Reply:
Gut, die Crypto-Software ist also schon als Bibliothekspaket in V2.14 enthalten.

Ich hatte den (Um-)Weg über pip und requirements.txt nicht verstanden und angenommen, jeder V2.14 Anwender muss hier selbst einsteigen und pycarwings manuell installieren.

Jetzt fehlt nach wie vor noch eine Ausprobe des Leaf-Moduls im V2.14 Umfeld.
ddraeyer schlägt unten die Ausprobe eines V2.14 Abbildes auf einer VM mit Debian 11 vor.
Wäre das ausreichend? Oder muss der Test auf einer OpenWB mit RPi und V2.14 laufen?

Reply:
Hallo @mekrapp 

Ausreichend ist es nicht. Sobald es jedoch als auf der VM als lauffähig gemeldet ist, werde ich sofort auf dem RasPi mit openWB 2.14 einen Test durchführen. Bitte jetzt auf der VM bis zur Lauffähigkeit entwickeln.